### ISSUE TITLE
AttributeError: 'Publication' object has no attribute 'citations_link'

### ISSUE THREAD

**Issue by ipeirotis** [scholarly-python-package/scholarly#168]
The example from the documentation fails:

```python
from scholarly import scholarly

# Retrieve the author's data, fill-in, and print
search_query = scholarly.search_author('Steven A Cholewiak')
author = next(search_query).fill()
print(author)

# Print the titles of the author's publications
print([pub.bib['title'] for pub in author.publications])

# Take a closer look at the first publication
pub = author.publications[0].fill()
print(pub)

# Which papers cited that publication?
print([citation.bib['title'] for citation in pub.citedby])
```

The last command returns the following error:

```
/usr/local/lib/python3.8/dist-packages/scholarly/publication.py in citedby(self)
    263         if not self.filled:
    264             self.fill()
--> 265         return _SearchScholarIterator(self.nav, self.citations_link)
    266 
    267     @property

AttributeError: 'Publication' object has no attribute 'citations_link'
```

**Comment by bronwencc** [comment]
I tried fixing this by adding a few lines to publication.py and renaming one definition of 'cites' to be 'cites_id' in `bib`.

In the `fill(self)` method:
`self.citations_link = _CITEDBYLINK.format(self.bib['cites_id'])`
At the top of the file with the other link formats, I also added:
`_CITEDBYLINK = '/scholar?cites={0}'`

This workaround renames one version of "cites" in `bib` to be "cites_id" because there were two different values being assigned to "cites". I changed one of them to "cites_id", specifically in the following if statement in the `fill(self)` method:
```
elif key == 'total citations':
                    self.bib['cites_id'] = re.findall(_SCHOLARPUBRE, val.a['href'])[0]
```
The other possible value for `bib['cites']` is the number of times the publication has been cited.

However, then the error is "Cannot fetch the page from Google Scholar." when trying `pub.citedby` but it looked more like an IP address issue (error code 429) than a `scholarly` issue since the URL that is generated with `"cites_id"` itself works.

E.g., With "Donna Strickland" for the name, I got [https://scholar.google.com/scholar?cites=14308308836162828619](https://scholar.google.com/scholar?cites=14308308836162828619) for the first publication's citations.
